HerPublicWebsite.UnitTests/BusinessLogic/Services/DateHelperTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using HerPublicWebsite.BusinessLogic.Services.RegularJobs;
+using NUnit.Framework;
+
+namespace Tests.BusinessLogic.Services;
+
+public class DateHelperTests
+{
+    [Test]
+    public void StartOfMonthService_WhenCalledWithoutParameter_ReturnsAPreviousDate()
+    {
+        // arrange
+        var startOfMonthService = new DateHelper();
+        
+        // act
+        var result = startOfMonthService.GetStartOfPreviousMonth();
+        
+        // assert
+        Assert.That(result <= DateTime.Today);
+    }
+    
+    [Test]
+    public void StartOfMonthService_WhenCalledWithoutParameter_ReturnsFirstDayOfAMonth()
+    {
+        // arrange
+        var startOfMonthService = new DateHelper();
+        
+        // act
+        var result = startOfMonthService.GetStartOfPreviousMonth();
+        
+        // assert
+        Assert.That(result.Day == 1);
+    }
+    
+    [Test]
+    public void StartOfMonthService_WhenCalledWithParameter_ReturnsAPreviousDate()
+    {
+        // arrange
+        var date = DateTime.Today;
+        var startOfMonthService = new DateHelper(() => date);
+        
+        // act
+        var result = startOfMonthService.GetStartOfPreviousMonth();
+        
+        // assert
+        Assert.That(result <= date);
+    }
+    
+    [TestCase("2024-10-12", "2024-9-1")]
+    [TestCase("2024-9-1", "2024-8-1")]
+    [TestCase("2024-2-29", "2024-1-1")] // Leap day tests
+    [TestCase("2024-3-29", "2024-2-1")] // Leap day tests
+    [TestCase("2024-4-30", "2024-3-1")] // End of differently sized month
+    [TestCase("2024-5-31", "2024-4-1")] // End of differently sized month
+    [TestCase("2024-8-31", "2024-7-1")] // End of same sized month
+    [TestCase("2025-1-31", "2024-12-1")] // Across year boundary
```

Comment:
It's much more typical to see dates written with leading zeroes in the month and day places. So `2024-09-01`, etc.